pkg/virtctl/create/instancetype/instancetype_test.go

```diff
@@ -0,0 +1,193 @@
+				setFlag(CPUFlag, "2"),
+				setFlag(MemoryFlag, "256Mi"),
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			instancetypeSpec = getInstancetypeSpec(namespaced, bytes)
+			Expect(instancetypeSpec.CPU.Guest).To(Equal(uint32(2)))
+			Expect(instancetypeSpec.Memory.Guest).To(Equal(resource.MustParse("256Mi")))
+		},
+			Entry("VirtualMachineInstancetype", namespaced),
+			Entry("VirtualMachineClusterInstancetype", ""),
+		)
+
+		DescribeTable("should succeed with defined gpus", func(namespaced string) {
+			bytes, err := clientcmd.NewRepeatableVirtctlCommandWithOut(create, Instancetype, namespaced,
+				setFlag(CPUFlag, "1"),
+				setFlag(MemoryFlag, "128Mi"),
+				setFlag(GPUFlag, "name:gpu1,devicename:nvidia"),
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			instancetypeSpec = getInstancetypeSpec(namespaced, bytes)
+			Expect(instancetypeSpec.GPUs).To(HaveLen(1))
+			Expect(instancetypeSpec.GPUs[0].Name).To(Equal("gpu1"))
+			Expect(instancetypeSpec.GPUs[0].DeviceName).To(Equal("nvidia"))
+		},
+			Entry("VirtualMachineInstancetype", namespaced),
+			Entry("VirtualMachineClusterInstancetype", ""),
+		)
+
+		DescribeTable("should succeed with defined hostDevices", func(namespaced string) {
+			bytes, err := clientcmd.NewRepeatableVirtctlCommandWithOut(create, Instancetype, namespaced,
+				setFlag(CPUFlag, "1"),
+				setFlag(MemoryFlag, "128Mi"),
+				setFlag(HostDeviceFlag, "name:device1,devicename:intel"),
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			instancetypeSpec = getInstancetypeSpec(namespaced, bytes)
+			Expect(instancetypeSpec.HostDevices).To(HaveLen(1))
+			Expect(instancetypeSpec.HostDevices[0].Name).To(Equal("device1"))
+			Expect(instancetypeSpec.HostDevices[0].DeviceName).To(Equal("intel"))
+		},
+			Entry("VirtualMachineInstancetype", namespaced),
+			Entry("VirtualMachineClusterInstancetype", ""),
+		)
+
+		DescribeTable("should succeed with valid IOThreadsPolicy", func(namespaced, param string, policy v1.IOThreadsPolicy) {
+			bytes, err := clientcmd.NewRepeatableVirtctlCommandWithOut(create, Instancetype, namespaced,
+				setFlag(CPUFlag, "1"),
+				setFlag(MemoryFlag, "128Mi"),
+				setFlag(IOThreadsPolicyFlag, param),
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			instancetypeSpec := getInstancetypeSpec(namespaced, bytes)
+			Expect(*instancetypeSpec.IOThreadsPolicy).To(Equal(policy))
+
+		},
+			Entry("VirtualMachineInstacetype set to auto", namespaced, "auto", v1.IOThreadsPolicyAuto),
+			Entry("VirtualMachineInstacetype set to shared", namespaced, "shared", v1.IOThreadsPolicyShared),
+
+			Entry("VirtualMachineClusterInstacetype set to auto", "", "auto", v1.IOThreadsPolicyAuto),
+			Entry("VirtualMachineClusterInstacetype set to shared", "", "shared", v1.IOThreadsPolicyShared),
+		)
+
+		DescribeTable("invalid cpu and memory", func(cpu, memory, errMsg string) {
+			err := clientcmd.NewRepeatableVirtctlCommand(create, Instancetype, namespaced,
```

Comment:
`_, err :=`

Reply:
This function returns only error from command execution.